### DISCUSSION TITLE
Stop a pipeline programmatically

### DISCUSSION THREAD

**Discussion by schergui-thefork** [kedro-org/kedro#1727] [Idea]
Hello,

I have a hook that runs a data check and I would like to know what's the **best way to stop the currently running pipeline** as well as stop the other pipelines from being executed programmatically (I want to stop the execution of kedro from within python if the data quality checks are not ok)

Thanks

**Comment by noklam** [comment]
@schergui-thefork For that you can simply raise an Exception, it may be more appropriate to have your custom exception such as `DataQualityCheckFailureError` etc. Or if you are using libraries like `great-expectations` or `pandera` they probably have the Exception class in place already.

**Reply by antonymilne** [reply]
FYI if you want to put some custom behaviour in when the exception is raised, you could use the `on_node_error` or `on_pipeline_error` hooks. These take an argument `error` so you could do `if isinstance(error, DataQualityCheckFailureError)` and put custom logic in there.
